Modal Component (#2795)
sumup-oss/circuit-ui: packages/circuit-ui/components/Modal/Modal.mdx

Reviewer: I've taken the liberty of editing this file directly (6c228816b91a483bef8717d11ba82432ce4c26f1). This seemed easier than commenting a bunch of nitpicks.

# Modal

<Status variant="stable" />

> The dialog component displays self-contained tasks in a modal view, requiring the user to interact with it before returning to the underlying content.

Reviewer: ```suggestion
The modal component displays self-contained tasks in an overlay view, requiring the user to interact with it before returning to the underlying content.
```

The modal component displays self-contained tasks in an overlay view, requiring the user to interact with it before returning to the underlying content.

<Story of={Stories.Base} />
<Props />

## When to use it

Generally, use the modal component sparingly. Consider displaying more complex tasks and large amounts of information on a separate page instead.
Generally, use the modal dialog component sparingly. Consider displaying more complex tasks and large amounts of information on a separate page instead.

## Variants
A modal dialog is a disruptive pattern that interrupts the user's current task. It should only be used when the task requires immediate attention or when the user needs to make a decision before continuing:

<Story of={Stories.Variants} />
- Confirming a critical action, such as deleting an item or abandoning a task (also see the [NotificationModal](Notification/NotificationModal/Docs) component).
- Forms or inputs to collect small amounts of data without navigating away from the current page (e.g., login, feedback forms).
- Focused tasks like editing an item, uploading files, or reviewing details.

> ### Contextual

Reviewer: What happened to the contextual variant (docs)?

Reviewer: Even though it's the default variant, we need to document the `contextual` variant to help users choose between the variants.
